crates/stc_ts_file_analyzer/src/analyzer/stmt/loops.rs

```diff
@@ -89,7 +90,16 @@ impl Analyzer<'_, '_> {
 
         self.cur_facts.true_facts += prev_facts;
         self.cur_facts.false_facts += prev_false_facts;
-
+        if !self.scope.return_values.in_conditional {
+            let mut v = LoopBreakerFinder { found: false };
+            body.visit_with(&mut v);
+            let has_break = v.found;
+            if !has_break {
+                if let Some(RExpr::Lit(RLit::Bool(RBool { value, .. }))) = test {
+                    self.ctx.in_unreachable = *value;
```

Comment:
Isn't this `!*value`?

`while (true)` => reachable
`while (false)` => unreachable

Reply:
`while true` is unreachable

Reply:
if infinite loop in scope, we can get unreachable
infiinite loop comes from `condition statement` is always `true`

Reply:
Ah, I see. Thanks!